novawallet/Modules/Staking/NominationPools/BondMore/Base/NominationPoolBondMoreBaseInteractor.swift

```diff
@@ -0,0 +1,392 @@
+    }
+
+    func createExtrinsicClosure(for points: BigUInt) -> ExtrinsicBuilderClosure {
+        { builder in
+            let call = NominationPools.BondExtraCall(extra: .freeBalance(points))
+            return try builder.adding(call: call.runtimeCall())
+        }
+    }
+
+    func subscribePoolProviders() {
+        guard let poolId = currentPoolId else {
+            return
+        }
+
+        bondedPoolProvider = subscribeBondedPool(for: poolId, chainId: chainId)
+        rewardPoolProvider = subscribeRewardPool(for: poolId, chainId: chainId)
+
+        subscribeClaimableRewardsProvider()
+    }
+
+    func subscribeClaimableRewardsProvider() {
+        guard let poolId = currentPoolId else {
+            return
+        }
+
+        claimableRewardProvider = subscribeClaimableRewards(
+            for: chainId,
+            poolId: poolId,
+            accountId: accountId
+        )
+
+        if claimableRewardProvider == nil {
+            basePresenter?.didReceive(error: .claimableRewards(CommonError.dataCorruption))
+        }
+    }
+
+    func subscribePoolMember() {
+        clear(dataProvider: &poolMemberProvider)
+        poolMemberProvider = subscribePoolMember(for: accountId, chainId: chainId)
+    }
+
+    func provideAssetExistence() {
+        let assetInfoWrapper = assetStorageInfoFactory.createStorageInfoWrapper(
+            from: chainAsset.asset,
+            runtimeProvider: runtimeService
+        )
+
+        let wrapper: CompoundOperationWrapper<AssetBalanceExistence?> =
+            OperationCombiningService.compoundWrapper(operationManager: operationManager) { [weak self] in
+                guard let self = self else {
+                    return nil
+                }
+                let assetInfo = try assetInfoWrapper.targetOperation.extractNoCancellableResultData()
+
+                return self.assetStorageInfoFactory.createAssetBalanceExistenceOperation(
+                    for: assetInfo,
+                    chainId: chainAsset.chain.chainId,
+                    asset: chainAsset.asset
+                )
+            }
+
+        wrapper.targetOperation.completionBlock = { [weak self] in
+            DispatchQueue.main.async {
+                guard self?.assetExistenceCancellable === wrapper else {
+                    return
+                }
+                self?.assetExistenceCancellable = nil
+
+                do {
+                    let assetExistence = try wrapper.targetOperation.extractNoCancellableResultData()
+                    self?.basePresenter?.didReceive(assetBalanceExistance: assetExistence)
+                } catch {
+                    self?.basePresenter?.didReceive(error: .assetExistance(error))
+                }
+            }
+        }
+
+        wrapper.addDependency(wrapper: assetInfoWrapper)
+        assetExistenceCancellable = wrapper
+        operationQueue.addOperations(wrapper.allOperations, waitUntilFinished: false)
```

Comment:
Looks like you don't run assetInfoWrapper. Also I would suggest to create a single wrapper from two you have and save it as cancellable.